### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by duburcqa**
- [jiminy core] Add jointType attribute to the motor. Only support linear and rotary joints for motors.
- [jiminy core] Replace Torque by Effort to be appropriate for both rotary and linear joints.
- [jiminy core] Add prefix 'joints' to the ambiguous fields (Position|Velocity)Limit*.
- [jiminy core] Define more appropriate position/velocity/effort limits for non-physical joints.

____

- [jiminy_py C++] Improve pinocchio/eigenpy 'submodule' management.
- [jiminy_py C++] Add getters and expose sensor type specific properties. Fix wrong exposure of the sensors names.
- [jiminy_py C++] Fix Robot class used instead of Model in Jiminy bindings.

____

- [jiminy_py Python] Improve robustness of viewer to handle closed gepetto-gui during rendering.
- [jiminy_py Python] Enable to render the initial state of EngineAsynchronous.

____

- [gym_jiminy] Refactoring of Gym Jiminy to extract more information from the model instead of using hardcoded values.
-  [gym_jiminy] The observation space now corresponds to the sensors and state spaces by default.
-  [gym_jiminy] Use Global variable to define "universal bounds" to overwrite jiminy inf bounds if desired. 

____

- [jiminy_plot] Move plotter.py in log.py since they are related. Remove Python3 only typing.
- [jiminy_plot] Prepend expressions with ':' enable to plot every matched fields in the same subplot.

For example
```
jiminy_plot /tmp/traj.data :*.FX:*.FY:*.FZ :*.Q :*.V *energy
```

image
____

- [doc] Update readme.

### CODE REVIEW

**Comment by duburcqa** [comment]
ping @matthieuvigne

**Comment by matthieuvigne** [comment]
Do you really want to support python2.7 now that it's officially deprecated ?  Removing typing hints because a deprecated version of python doesn't support it seems kind of weird...

File: python/jiminy_py/src/jiminy_py/log.py [hunk 1]
```diff
@@ -81,14 +97,79 @@ def read_log(filename:str) -> tp.Tuple[tp.Dict, tp.Dict]:
...
+    if len(plotting_commands) == 0:
+        print("Available data:")
+        print("\n - ".join(log_data.keys()))
+        exit(0)
+
+    # Parse plotting arguments.
+    plotted_elements = []
+    for cmd in plotting_commands:
+        # Check that the command is valid, i.e. that all elements exits. If it is the case, add it to the list.
+        same_subplot = (cmd[0] == ':')
```
Comment (matthieuvigne) [inline python/jiminy_py/src/jiminy_py/log.py:130]: Nice feature :)

**Comment by duburcqa** [comment]
> Do you really want to support python2.7 now that it's officially deprecated ? Removing typing hints because a deprecated version of python doesn't support it seems kind of weird...

I don't know. But we need to decide if we really want to completely drop the support officially or not, and if so, update the Readme. According to the statistics, between 5% and 15% of the researchers are still using Python2. Personally, I don't think typing is enough to drop support before the end of 2020. What do you think ?

**Comment by matthieuvigne** [comment]
No I guess not - the best would be to have a separate python2 and python3 version, but clearly we don't want to maintain that. So let's stick to python2 for now...

**Pull request merged**